src/main/java/com/example/demo/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.example.demo.repository.impl;
+
+import com.example.demo.exception.DataProcessingException;
+import com.example.demo.model.Book;
+import com.example.demo2.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import java.util.List;
+
+@Repository
```

Comment:
It's better to use Lombok annotation
```suggestion
@Repository
@RequiredArgsConstructor
```